File: Data Structures & Algorithms/baseball-game/submission-2.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        self.stack = []

        actions_dict = {
            "C": self.stack.pop,
            "D": self.double,
            "+": self.sums,
        }

        for op in operations:
            if op in actions_dict:
                actions_dict[op]()
            else:
                self.stack.append(int(op))
        
        total = sum(self.stack)

        return total
    
    def double(self):
        self.stack.append(2*self.stack[-1])
    
    def sums(self):
        self.stack.append(self.stack[-2] + self.stack[-1])
            
            

File: Data Structures & Algorithms/validate-parentheses/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        const dict = {
            '(':')',
            '{':'}',
            '[':']',
        };
        const stack = [];
        for (const char of s) {
            if (char in dict) stack.push(dict[char]);
            else {
                if(stack.at(-1) !== char) return false;
                stack.pop();
            }
        }
        return stack.length === 0;
    }
}
